packages/medusa/src/services/shipping-option.js

```diff
@@ -287,6 +295,23 @@ class ShippingOptionService extends BaseService {
     })
   }
 
+  /**
+   * Creates a shipping method for a given cart.
+   * @param {string} optionId - the id of the option to use for the method.
+   * @param {string} cartId - the cart for which the shipping option is updated.
+   * @param {object} update - the update which is applied to the cart.
+   * @return {Promise<ShippingMethod>} the resulting shipping method.
+   */
+  async updateShippingMethodForCart(optionId, cartId, update) {
```

Comment:
Don't like the naming here, since we already have an `updateShippingMethod` (for carts) in the option service. Could we either create a `fetchByOptionAndCart` and use that with `updateShippingMethod` in the route, or rename this to differentiate the two methods.